pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArithmeticFunctions.java

```diff
@@ -104,4 +111,30 @@ public static double ln(double a) {
   public static double sqrt(double a) {
     return Math.sqrt(a);
   }
+
+  @ScalarFunction
+  public static double sign(double a) {
+    return Math.signum(a);
+  }
+
+  @ScalarFunction
+  public static double pi() { return Math.PI; }
+
+  @ScalarFunction
+  public static double e() { return Math.E; }
+
+  @ScalarFunction
+  public static double power(double a, double b) {
+    return Math.pow(a, b);
+  }
+
+  @ScalarFunction
+  public static double log(double a) {
+    return Math.log10(a);
+  }
```

Comment:
@richardstartin Agree that `ScalarFunction` is not as efficient as `TransformFunction`, but we need `ScalarFunction` for ingestion transform and post-aggregation. We can add `ScalarFunction` for functionality first, then add `TransformFunction` for performance optimization later.

Reply:
+1

Reply:
I understand that it would be _nice_ to have these available for ingestion transforms, but that isn't actually a SQL interface and those functions don't need to have the names used in SQL (though for simplicity it would be ideal if they did match!), so if the goal of adding the functions at all is to increase SQL compliance I would prioritise usage in SQL, which means implementing TransformFunctions, given the huge performance gap. 

Support for ingestion transform configs can be added later when it's possible to do it without making queries inefficient. Moreover, once something has been added, it can't be removed, so queries relying on `@ScalarFunction` will be difficult to optimise without major rework.

Reply:
This is also required for post-aggregation (etc. `SELECT log10(sum(col1)) FROM myTable`) which is part of SQL, so we need to ensure the scalar function and transform function matches.

So I'd suggest doing it following these steps:
1. Add the scalar function version so that the query works (performance won't be very good without proper vectorization)
2. Add the transform function version to optimize the performance

Reply:
On it.

Reply:
Added transform functions classes for Arithmetic and Trigonometric functions.

Reply:
@mneedham I had discussion with @Jackie-Jiang in which he confirmed that Pinot prioritises TransformFunctions first and if not found, it will use ScalarFunctions. Both can have the same name. The following piece of code in TransformFunctionFactory ensures this order.

```
  TransformFunction transformFunction;
        Class<? extends TransformFunction> transformFunctionClass = TRANSFORM_FUNCTION_MAP.get(functionName);
        if (transformFunctionClass != null) {
          // Transform function
          try {
            transformFunction = transformFunctionClass.newInstance();
          } catch (Exception e) {
            throw new RuntimeException("Caught exception while constructing transform function: " + functionName, e);
          }
        } else {
          // Scalar function
          FunctionInfo functionInfo = FunctionRegistry.getFunctionInfo(functionName, numArguments);
          if (functionInfo == null) {
            if (FunctionRegistry.containsFunction(functionName)) {
              throw new BadQueryRequestException(
                String.format("Unsupported function: %s with %d parameters", functionName, numArguments));
            } else {
              throw new BadQueryRequestException(
                String.format("Unsupported function: %s not found", functionName));
            }
          }
          transformFunction = new ScalarTransformFunctionWrapper(functionInfo);
        }
```